tests/apollo/test_skvbc_preexecution.py

```diff
@@ -191,3 +192,41 @@ async def test_view_change(self, bft_network):
                 break
 
         self.assertEqual(new_last_block, last_block)
+
+    @with_trio
+    @with_bft_network(start_replica_cmd)
+    async def test_f_nonprimary_crash_amid_parallel_tx(self, bft_network):
+        '''
+        Crash f nonprimary replicas while multiple transactions are being processed.
+        Block processing of the network should be unaffected with f-count interruption.
+        Check created block lengths match submitted x count of transactions.
+        '''
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        f_count = int((len(bft_network.all_replicas()) - 1) / 3)
+        primary_index = 0
+
+        await trio.sleep(2)
+        
+        num_of_requests = 20
+        clients = bft_network.random_clients(1)
+        reader_client = list(clients)[0]
+        nonprimaries = bft_network.all_replicas(without={primary_index})
+        crash_targets = random.sample(nonprimaries, f_count) # pick random f to crash
+        
+        started_block_count = skvbc.parse_reply(await reader_client.read(skvbc.get_last_block_req()))
+        with trio.move_on_after(seconds=20): # around tx_count seconds for enough buffer time
+            async_requests = self.run_concurrent_pre_execution_requests(
+                                            skvbc, clients, num_of_requests,
+                                            write_weight=1.0, interval=0.05)
+            await trio.sleep(random.uniform(0.5, 2.5)) # slight delay to make parallel tx get in to process
+            bft_network.stop_replicas(crash_targets) # crash chosen replicas in the middle of tx handling
```

Comment:
I suggest to do this before sending the concurrent requests. The way you've implemented it now, you first send requests and then stop the replicas, which isn't very useful...

Reply:
Assigning with `async_requests = self.run_concurrent_pre_execution_requests` (note, it's not yet awaited when called) and awaiting `async_requests` AFTER stop_replicas call will stop replicas in the middle of other coroutine executing `run_concurrent_pre_execution_requests`.

Reply:
For the sake of deterministic test execution, let's please simplify and crash the f replicas before sending the requests. Thank you.

Reply:
PS: if you want to do both in parallel, you should read about nurseries, and implement this scenario as a separate test

Reply:
Certainly. That would be 100% deterministic, and simpler.

I just assumed that this was in-the-middle crash from BC-2534 title ([Apollo] Crash up to f non-primary replicas while running multiple pre-execution requests)